tests/tpch/generate-data.py

```diff
@@ -180,9 +216,15 @@ def rows_approx_mb(con, table_name, partition_size: str):
     table = con.sql(f"select * from {table_name} limit {sample_size}").arrow()
 
     with tempfile.TemporaryDirectory() as tmpdir:
-        tmp = pathlib.Path(tmpdir) / "out.parquet"
-        con.sql(
-            f"copy (select * from {table_name} limit {sample_size}) to '{tmp}' (format parquet)"
+        tmp = pathlib.Path(tmpdir) / "tmp.parquet"
+        stmt = f"select * from {table_name} limit {sample_size}"
+        df = con.sql(stmt).arrow()
+        pq.write_table(
+            df,
+            tmp,
+            compression=compression.value.lower(),
+            write_statistics=True,
+            write_page_index=True,
```

Comment:
`pyarrow` doesn't store page indices by default and I'm not even sure if it is implemented to use them during reading.
Whether this is a good idea to have depends on many different things, size of file, size of row groups, number of columns, etc. and for some constellations this can cause overhead.
Before we enable this blindly, I would like to make sure this does not negatively impact anything.

Reply:
I see a small difference in file size (see metadata below) when adding page index, and no real performance difference:

```bash
❯ ls -lhs lineitem*                                                                                                         (tmp)
126M -rw-r--r--. 1 milesg milesg 126M Nov 28 08:47 lineitem.duckdb.snappy.parquet
102M -rw-r--r--. 1 milesg milesg 102M Nov 28 12:09 lineitem-stats-no-index.snappy.parquet
102M -rw-r--r--. 1 milesg milesg 102M Nov 30 11:53 lineitem-stats-and-index.snappy.parquet
```

```bash
❯ hyperfine 'python read-no-page-index.py' 'python read-page-index.py'                                                      (tmp)
Benchmark 1: python read-no-page-index.py
  Time (mean ± σ):      1.299 s ±  0.160 s    [User: 2.656 s, System: 2.409 s]
  Range (min … max):    0.994 s …  1.572 s    10 runs

Benchmark 2: python read-page-index.py
  Time (mean ± σ):      1.272 s ±  0.159 s    [User: 2.514 s, System: 2.381 s]
  Range (min … max):    1.084 s …  1.538 s    10 runs

Summary
  python read-page-index.py ran
    1.02 ± 0.18 times faster than python read-no-page-index.py
```

Then metadata:

```python
In [5]: pq.ParquetFile('lineitem-stats-no-index.snappy.parquet').metadata
Out[5]:
<pyarrow._parquet.FileMetaData object at 0x7fc36c2da520>
  created_by: parquet-cpp-arrow version 13.0.0
  num_columns: 16
  num_rows: 2735597
  num_row_groups: 3
  format_version: 2.6
  serialized_size: 6900

In [6]: pq.ParquetFile('lineitem-stats-and-index.snappy.parquet').metadata
Out[6]:
<pyarrow._parquet.FileMetaData object at 0x7fc36c4f4fe0>
  created_by: parquet-cpp-arrow version 13.0.0
  num_columns: 16
  num_rows: 2735597
  num_row_groups: 3
  format_version: 2.6
  serialized_size: 7584
```